cci_nepal/pipeline/dummy_data/model_complete.py

```diff
@@ -0,0 +1,367 @@
+df.head()
+
+# #### Creating Additional Features ####
+
+# In this section, we will create 3 additional features combined out of existing features to see how effectively they will be in Model. The 2 newly created features will be 'Total Members', 'Total Female' and 'Total Children'
+
+df.insert(2, "Total_Members", df.iloc[:, 7:28].sum(axis=1))
+df.insert(3, "Total_Female", df.iloc[:, 17:27].sum(axis=1))
+# df.insert(4, "Total_Children", df.iloc[:, [7, 8, 9, 17, 18, 19]].sum(axis=1))
+
+# #### Mapping Categories to Numbers ####
+
+# The idea is to map categorical labels of NFRI Preferences (Essential, Desirable, Unnecessary) into numeric scores to be used in Model. That way, we will have a continuous scale for NFRI Importance Score.
+
+mapping = {"Essential": 3, "Desirable": 2, "Unnecessary": 1}
+df_clean = df.applymap(lambda s: mapping.get(s) if s in mapping else s)
+
+df_clean.head(5)
+
+# #### Test Section Only. To be excluded for real data ####
+
+# Since the randomly created labels are somewhat evenly generated, the predicted label mostly averages out as 2 (Desirable). Thus, to ensure our Model can predict all three labels, we are feeding the score of 1 and 3 to all observations of some of the items here.
+
+ddu.replace_column_with_number(
+    df_clean, ["Blanket", "Printed Cloth", "Hand Sanitizer"], [3, 1, 3]
+)
+
+df_clean.head(3)
+
+# #### Categorizing appropriate variables ####
+
+# The following code is to first automatically collect all the categorical columns from the dataset, and then categorize the columns as a preparatory step for Linear Regression model.
+
+# +
+num_cols = df_clean._get_numeric_data().columns
+
+columns_to_categorize = list(set(df_clean.columns) - set(num_cols))
+
+df_clean[columns_to_categorize] = df_clean[columns_to_categorize].astype("category")
+# -
+
+model_basic = LinearRegression()
+model_non_basic = LinearRegression()
+
+# #### Collecting Input and Output features ####
+
+input_features = df_clean.columns[:-21]
+output_features = df_clean.columns[-21:]
+input_numerical_features = [
+    features for features in input_features if features not in columns_to_categorize
+]
+
+# #### Feature Scaling ####
+
+# In this step, we will form Feature Scaling of numerical input features using Min-Max scaler.
+
+scaler = MinMaxScaler()
+df_clean[input_numerical_features] = scaler.fit_transform(
+    df_clean[input_numerical_features]
+)
+
+# #### Feature Selection Step ####
+
+# This is the section where we will manually select the Features to be kept into the Model. For real data, we will perform Feature Analysis, Selection and Engineering process before this step. For now, we will select all the Features.
+
+X = df_clean[input_features]  # Collecting all features except the NFRI Item Preferences
+X = pd.get_dummies(data=X, drop_first=True)
+
+# Dividing NFRI into basic and non-basic in the step below. So as to prepare two separate Model for basic and non-basic items.
+
+# +
```

Comment:
I wonder if we want these as small text files? Or one dictionary which holds both (json file). Just think it might be good to have them somewhere easy to find as they are quite nestled in the code.